pilot/pkg/model/service.go

```diff
@@ -608,9 +608,9 @@ func ParseSubsetKey(s string) (direction TrafficDirection, subsetName string, ho
 }
 
 // GetServiceAddressForProxy returns a Service's IP address specific to the cluster where the node resides
-func (s *Service) GetServiceAddressForProxy(node *Proxy, push *PushContext) string {
-	if node.Metadata != nil && node.Metadata.ClusterID != "" && push.ServiceIndex.ClusterVIPs[s][node.Metadata.ClusterID] != "" {
-		return push.ServiceIndex.ClusterVIPs[s][node.Metadata.ClusterID]
+func (s *Service) GetServiceAddressForProxy(node *Proxy) string {
+	if node.Metadata != nil && node.Metadata.ClusterID != "" && s.ClusterVIPs[node.Metadata.ClusterID] != "" {
+		return s.ClusterVIPs[node.Metadata.ClusterID]
```

Comment:
Isn't ClusterVIP access need to be protected? I think that was original motivation for having mutex in service and GetServiceAddressForProxy used to hold a lock for every service on every push - So moved it to push context. Has some thing changed now?

Reply:
Yeah, need lock here.